lib/plugins-data.ts

```diff
@@ -55,24 +55,51 @@ export class CordovaPluginData implements IPlugin {
 
 	private getPluginVariablesInfo(): string[] {
 		let result: string[] = [];
-		_.each(this.configurations, (configuration: string) => {
-			let pluginVariablesData = this.$project.getProperty(this.$projectConstants.CORDOVA_PLUGIN_VARIABLES_PROPERTY_NAME, configuration);
+		if(this.configurations && this.configurations.length) {
+			_.each(this.configurations, (configuration: string) => {
+				let info = this.getPluginVarsStringInformation(configuration).wait();
+				result.push(...info);
+			});
+		} else {
+			let info = this.getPluginVarsStringInformation().wait();
+			result.push(...info);
+		}
+
+		return result;
+	}
+
+	private getPluginVarsStringInformation(configuration?: string): IFuture<string[]> {
+		return ((): string[] => {
+			let result: string[] = [];
+			let configString = configuration ? ` for ${configuration} configuration` : "";
+			let pluginVariablesData = this.$project.getPluginVariablesInfo(configuration).wait();
 			if(pluginVariablesData && pluginVariablesData[this.data.Identifier]) {
 				let variables = pluginVariablesData[this.data.Identifier];
 				let variableNames = _.keys(variables);
 				if(variableNames.length > 0) {
 					let output:string[] = [];
-					output.push(util.format("    Variables for %s configuration:", configuration));
+					output.push(`    Variables${configString}:`);
 					_.each(variableNames, (variableName:string) => {
 						output.push(util.format("        %s: %s", variableName, variables[variableName]));
 					});
 
 					result.push(output.join(EOL));
 				}
+			} else {
+				if(this.data.Variables) {
+					// We should never get here with anything that is not array, but anyway, lets assure we'll not throw some unexpected error.
+					if(_.isArray(this.data.Variables) && (<string[]>this.data.Variables).length) {
```

Comment:
I think it can be expressed more elegantly as
`if (this.data.Variables instanceof Array && this.data.Variables.length)` (see [type guards](https://github.com/Microsoft/TypeScript/wiki/What's-new-in-TypeScript#type-guards)